src/bp/ui-shared/src/ContextMenu/index.tsx

```diff
@@ -0,0 +1,161 @@
+                    popoverClassName={popoverClassName}
+                    target={<div />}
+                    transitionDuration={TRANSITION_DURATION}
+                />
+            </div>
+        );
+    }
+
+    public show(menu: JSX.Element, offset: IOffset, onClose?: () => void, isDarkTheme?: boolean) {
+        this.setState({ isOpen: true, menu, offset, onClose, isDarkTheme });
+    }
+
+    public hide() {
+        safeInvoke(this.state.onClose);
+        this.setState({ isOpen: false, onClose: undefined });
+    }
+
+    private cancelContextMenu = (e: React.SyntheticEvent<HTMLDivElement>) => e.preventDefault();
+
+    private handleBackdropContextMenu = (e: React.MouseEvent<HTMLDivElement>) => {
+        // React function to remove from the event pool, useful when using a event within a callback
+        e.persist();
+        e.preventDefault();
+        // wait for backdrop to disappear so we can find the "real" element at event coordinates.
+        // timeout duration is equivalent to transition duration so we know it's animated out.
+        this.setTimeout(() => {
+            // retrigger context menu event at the element beneath the backdrop.
+            // if it has a `contextmenu` event handler then it'll be invoked.
+            // if it doesn't, no native menu will show (at least on OSX) :(
+            const newTarget = document.elementFromPoint(e.clientX, e.clientY);
+            newTarget.dispatchEvent(new MouseEvent("contextmenu", e));
+        }, TRANSITION_DURATION);
+    };
+
+    private handlePopoverInteraction = (nextOpenState: boolean) => {
+        if (!nextOpenState) {
+            // delay the actual hiding till the event queue clears
+            // to avoid flicker of opening twice
+            requestAnimationFrame(() => this.hide());
+        }
+    };
+}
+
+let contextMenuElement: HTMLElement;
+let contextMenu: ContextMenu;
+*/
+import React, { FC, Fragment, SyntheticEvent, useEffect } from 'react'
+import ReactDOM from 'react-dom'
+
+import Overlay from '../Overlay'
+
+import style from './style.scss'
+
+const ContextMenuWrapper = ({ event, children }) => {
+  const elPos = event.currentTarget.getBoundingClientRect()
+
+  const handleToggle = e => {
+    e.stopPropagation()
+    removeContextMenu()
+  }
+
+  return (
+    <Fragment>
+      <div
+        style={{ top: `${elPos.top + elPos.height}px`, left: `${elPos.left + elPos.width / 2}px` }}
+        className={style.contextMenuWrapper}
+      >
+        {children}
+      </div>
+      <Overlay onClick={handleToggle} />
+    </Fragment>
+  )
+}
+const contextMenu = (e: SyntheticEvent, content: JSX.Element) => {
+  e.preventDefault()
+  e.stopPropagation()
+
+  const body = document.getElementsByTagName('body')[0]
+  const div = document.createElement('div')
+
+  div.setAttribute('id', 'context-menu-container')
+  body.appendChild(div)
+
+  ReactDOM.render(<ContextMenuWrapper event={e}>{content}</ContextMenuWrapper>, div)
```

Comment:
Could we have used a portal for that ?

Reply:
I started implementing it with portal actually, but I wanted to use a method call instead of rendering a component